### ISSUE TITLE
rustc does not detect CPU features correctly when target-cpu=native on Zen 3

### ISSUE THREAD

**Issue by as-com** [rust-lang/rust#80633]
I (actually, @tykosa, who managed to get a Ryzen 5 5600X) tried this code:

```rust
#[cfg(target_feature = "avx")]
fn foo() {
    println!("hi");
}

fn main() {
    foo();
}
```

I expected to see this happen: the compiler detects that the CPU has AVX support and compiles the code, like it does on other CPUs

Instead, this happened: 
```
$ rustc -C target-cpu=native test.rs
error[E0425]: cannot find function foo in this scope
 --> test.rs:7:9
  |
7 |         foo();
  |         ^^^ not found in this scope

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
```

### Meta

`rustc --version --verbose`:
```
rustc 1.51.0-nightly (17eec1433 2021-01-01)
binary: rustc
commit-hash: 17eec1433c69972844dd228b5fe801f218e118c3
commit-date: 2021-01-01
host: x86_64-unknown-linux-gnu
release: 1.51.0-nightly
```

`rustc --print=target-cpus`
```
Available CPUs for this target:
    native         - Select the CPU of the current host (currently generic).
    amdfam10
    athlon
    athlon-4
...
...
  yonah
    znver1
    znver2
```

Snippet of /proc/cpuinfo:
![image](https://user-images.githubusercontent.com/3819625/103469177-677d5200-4d2f-11eb-9010-565594f045fa.png)

**Comment by ehuss** [comment]
I believe zen 3 support will be included with LLVM 12, and rustc should pick it up when it upgrades to that release ([upstream change](https://reviews.llvm.org/rG2a136a7a9c68e4818b28bea6051b78a8181ba78e)).  I would guess that the 12 release is still a few months out.

**Comment by nagisa** [comment]
`target-cpu=native` is a best effort kind of feature and will not be able to detect 100% accurately in all instances. As thus `target-cpu=native` not detecting `znver3` is _not_ an implementation bug.

The only question is how can we better manage user's expectations wrt `-Ctarget-cpu=native`? This is not the first time this comes up.

I'll mark this as a duplicate of https://github.com/rust-lang/rust/issues/54688 for now.
